numpyro/contrib/tfp/mcmc.py

```diff
@@ -0,0 +1,233 @@
+                kernel = tfp.mcmc.MetropolisHastings(kernel)
+            self._init_fn, self._sample_fn = _extract_kernel_functions(kernel)
+        return init_params
+
+    @property
+    def model(self):
+        return self._model
+
+    @property
+    def sample_field(self):
+        return 'z'
+
+    @property
+    def default_fields(self):
+        return ('z',)
+
+    def get_diagnostics_str(self, state):
+        """
+        Given the current `state`, returns the diagnostics string to
+        be added to progress bar for diagnostics purpose.
+        """
+        return ''
+
+    def init(self, rng_key, num_warmup, init_params=None, model_args=(), model_kwargs={}):
+        # non-vectorized
+        if rng_key.ndim == 1:
+            rng_key, rng_key_init_model = random.split(rng_key)
+        # vectorized
+        else:
+            rng_key, rng_key_init_model = jnp.swapaxes(vmap(random.split)(rng_key), 0, 1)
+        init_params = self._init_state(rng_key_init_model, model_args, model_kwargs, init_params)
+        if self._potential_fn and init_params is None:
+            raise ValueError('Valid value of `init_params` must be provided with'
+                             ' `target_log_prob_fn`.')
+
+        if rng_key.ndim == 1:
+            init_state = self._init_fn(init_params, rng_key)
+        else:
+            # XXX it is safe to run hmc_init_fn under vmap despite that hmc_init_fn changes some
+            # nonlocal variables: momentum_generator, wa_update, trajectory_len, max_treedepth,
+            # wa_steps because those variables do not depend on traced args: init_params, rng_key.
+            init_state = vmap(self._init_fn)(init_params, rng_key)
+            sample_fn = vmap(self._sample_fn, in_axes=(0, None, None))
+            self._sample_fn = sample_fn
+        return init_state
+
+    def postprocess_fn(self, args, kwargs):
+        if self._postprocess_fn is None:
+            return identity
+        return self._postprocess_fn(*args, **kwargs)
+
+    def sample(self, state, model_args, model_kwargs):
+        """
+        Run the kernel from the given :data:`~numpyro.contrib.tfp.mcmc.TFPKernelState`
+        and return the resulting :data:`~numpyro.contrib.tfp.mcmc.TFPKernelState`.
+
+        :param TFPKernelState state: Represents the current state.
+        :param model_args: Arguments provided to the model.
+        :param model_kwargs: Keyword arguments provided to the model.
+        :return: Next `state` after running the kernel.
+        """
+        return self._sample_fn(state, model_args, model_kwargs)
+
+
+__all__ = ['TFPKernel']
+for _name, _Kernel in tfp.mcmc.__dict__.items():
+    if not isinstance(_Kernel, type):
+        continue
+    if not issubclass(_Kernel, tfp.mcmc.TransitionKernel):
+        continue
+    if 'target_log_prob_fn' not in inspect.getfullargspec(_Kernel).args:
+        continue
+
+    try:
+        _PyroKernel = locals()[_name]
```

Comment:
Since we aren't importing any numpyro mcmc kernels here, is this for a future use case?

Reply:
Thanks, it is unnecessary codes. I just remove it.